### ISSUE TITLE
How i am able to connect my device? 

### ISSUE THREAD

**Issue by DanielNowak98** [cryinkfly/VCDS-VAG-COM-Diagnosis-system-for-Linux#4]
Hey, thanks for your work! I would like to connect my device (HEX-V2) and would like to use it over wine. I run into the problem, that the device could not be found USB-Bibliothek: 03.02.07 

Do you know any way to fix this ? You got the device running ?

**Comment by cryinkfly** [comment]
@DanielNowak98 Unfortunately, I haven't been able to test it myself because I don't have the hardware to do it.

And are you a member of the following groups with your Linux user:

- lp
- tty
- dialout

More information about these groups can you find here: https://wiki.debian.org/SystemGroups

... because that's the only way you'll have access to the advice.

**Comment by DanielNowak98** [comment]
yes, i am member of your mentioned groups.

**Comment by PMKrol** [comment]
Few things to do to connect to Your car (despite installation script):
link your ttyUSBx device to com1 of wine.
chmod 777 ttyUSBx,
set baud to 9600 by stty.

I have script which fires up stty constantly. Works for me for most of subdevices - unfortunately not all.
I installed VCDS by PlayOnLinux.

**Comment by davidbitterlich** [comment]
Hi, I just wanted to give it a try with installing the software in wine and passthrough the hardware dongle via com. Unfortunately I cannot see the device as ttyUSB something. All I have in /dev is usb and vboxusb when it comes to USB... However the device is plugged in and I can see it via lsusb:

Bus 001 Device 002: ID 0403:fa24 Future Technology Devices International, Ltd MFT HEX-USB

Seems like my Kernel does not have a driver for it?

Well, in /sys/kernel/debug/usb/devices I can read the following regarding my device:
`S:  Product=MFT HEX-USB
S:  SerialNumber=RT000001
C:* #Ifs= 1 Cfg#= 1 Atr=80 MxPwr= 50mA
I:* If#= 0 Alt= 0 #EPs= 2 Cls=ff(vend.) Sub=ff Prot=ff Driver=(none)
E:  Ad=81(I) Atr=02(Bulk) MxPS=  64 Ivl=0ms
E:  Ad=02(O) Atr=02(Bulk) MxPS=  64 Ivl=0ms
`
Driver=(none), I can find an entry in /dev/bus/usb/001/002 but I guess this does not help me. 

I have no clue what's wrong on my side, I'm using Ubuntu with Kernel 6.5

May you tell more in detail how you managed to setup the connection?

Sincererly
David

**Comment by PMKrol** [comment]
List Your /dev before and after attaching USB. Check for differences.

________________________________
Od: David Bitterlich ***@***.***>
Wysłane: piątek, 29 marca 2024 11:16
Do: cryinkfly/VCDS-VAG-COM-Diagnosis-system-for-Linux ***@***.***>
DW: Patryk Król ***@***.***>; Comment ***@***.***>
Temat: Re: [cryinkfly/VCDS-VAG-COM-Diagnosis-system-for-Linux] How i am able to connect my device? (Issue #4)


Hi, I just wanted to give it a try with installing the software in wine and passthrough the hardware dongle via com. Unfortunately I cannot see the device as ttyUSB something. All I have in /dev is usb and vboxusb when it comes to USB... However the device is plugged in and I can see it via lsusb:

Bus 001 Device 002: ID 0403:fa24 Future Technology Devices International, Ltd MFT HEX-USB

Seems like my Kernel does not have a driver for it?

Well, in /sys/kernel/debug/usb/devices I can read the following regarding my device:
S: Product=MFT HEX-USB S: SerialNumber=RT000001 C:* #Ifs= 1 Cfg#= 1 Atr=80 MxPwr= 50mA I:* If#= 0 Alt= 0 #EPs= 2 Cls=ff(vend.) Sub=ff Prot=ff Driver=(none) E: Ad=81(I) Atr=02(Bulk) MxPS= 64 Ivl=0ms E: Ad=02(O) Atr=02(Bulk) MxPS= 64 Ivl=0ms
Driver=(none), I can find an entry in /dev/bus/usb/001/002 but I guess this does not help me.

I have no clue what's wrong on my side, I'm using Ubuntu with Kernel 6.5

May you tell more in detail how you managed to setup the connection?

Sincererly
David

**Comment by davidbitterlich** [comment]
That's what I did, but there were no differences. As the log from /sys/kernel/debug/usb/devices indicates, the Kernel does not know which driver it should use for this device and so it does not use any driver and therefore I do not have a device file in /dev directly (such as ttyUSB0). I took a look whether I could manually try to load a device driver for this device, but I did not have much success on this, yet.

**Comment by PMKrol** [comment]
According to this: https://forum.winehq.org/viewtopic.php?t=34864, it seems that it is not „normal” serial interface ;(.


dr inż. Patryk Król

Adiunkt
Assistant professor

Instytut Nauk Drzewnych i Meblarstwa
ul. Nowoursynowska 166, bud. 34, pok. 1/38, 02-787 Warszawa
tel. +48 22 593 85 76

Institute of Wood Sciences and Furniture
Nowoursynowska 166 St., bldg. 34, room 1/38, 02-787 Warsaw, Poland
ph. +48 22 593 85 76

SZKOŁA GŁÓWNA GOSPODARSTWA WIEJSKIEGO W WARSZAWIE
ul. Nowoursynowska 166, 02-787 Warszawa
www.sggw.edu.pl<http://www.sggw.pl/>

WARSAW UNIVERSITY OF LIFE SCIENCES
Nowoursynowska St. 166, 02-787 Warsaw, Poland
www.sggw.edu.pl/en/<http://www.sggw.edu.pl/en/>

Niniejsza wiadomość (wraz z załącznikiem) jest własnością Szkoły Głównej Gospodarstwa Wiejskiego w Warszawie (dalej: SGGW) i może zawierać informacje służbowe prawnie chronione. Jeśli nie są Państwo właściwym adresatem lub otrzymali Państwo tę wiadomość na skutek pomyłki, prosimy o tym fakcie niezwłocznie poinformować nadawcę i usunąć otrzymaną wiadomość. Każde nieautoryzowane kopiowanie, ujawnienie lub rozpowszechnianie załączonej informacji jest zabronione.
SGGW będzie Administratorem Danych Osobowych w rozumieniu przepisów Ogólnego Rozporządzenia o Ochronie Danych Osobowych (UE) 2016/679 z dnia 27 kwietnia 2016 r. (RODO) w związku z prowadzoną korespondencją, co stanowi jej prawnie uzasadniony interes, zgodnie z art. 6 ust. 1 lit. f) RODO Na SGGW został wyznaczony Inspektor Ochrony Danych, z którym kontakt możliwy jest pod adresem e-mail: ***@***.******@***.***> w sprawach przetwarzania danych osobowych.
Ma Pani/Pan prawo dostępu do swoich danych, możliwość ich sprostowania, usunięcia, a także wniesienia usprawiedliwionego sprzeciwu lub ograniczenia ich przetwarzania. Może Pani/Pan również złożyć skargę do Prezesa Urzędu Ochrony Danych Osobowych. Więcej informacji o przetwarzaniu Pani/Pana danych osobowych.<http://informacjarodo.sggw.edu.pl/>

This message (with the attachment) is the property of the Warsaw University of Life Sciences (SGGW) and may contain legally protected official information. If you are not the correct recipient or if you have received this message due to a mistake, please inform the sender immediately and delete the message. Any unauthorized copying, disclosure or dissemination of the accompanying information is prohibited.
SGGW will be the Administrator of Personal Data within the meaning of the General Data Protection Regulation (EU) 2016/679 of 27 April 2016 (GDPR) in connection with correspondence, which is its legitimate interest, in accordance with art. 6 sec. 1 lit. f) GDPR. SGGW. The Data Protection Officer has been appointed at SGGW, who can be contacted at the following e-mail address: ***@***.*** about matters related to personal data processing.
You have the right to access your data, rectify it, delete it, and also raise a justified objection or limit its processing. You can also submit a complaint to the President of the Personal Data Protection Office. More information on the processing of your personal data<http://informacjarodo.sggw.edu.pl/>.

________________________________
Od: David Bitterlich ***@***.***>
Wysłane: wtorek, 2 kwietnia 2024 15:45
Do: cryinkfly/VCDS-VAG-COM-Diagnosis-system-for-Linux ***@***.***>
DW: Patryk Król ***@***.***>; Comment ***@***.***>
Temat: Re: [cryinkfly/VCDS-VAG-COM-Diagnosis-system-for-Linux] How i am able to connect my device? (Issue #4)


That's what I did, but there were no differences. As the log from /sys/kernel/debug/usb/devices indicates, the Kernel does not know which driver it should use for this device and so it does not use any driver and therefore I do not have a device file in /dev directly (such as ttyUSB0). I took a look whether I could manually try to load a device driver for this device, but I did not have much success on this, yet.

**Comment by davidbitterlich** [comment]
Yes you're right and it's pretty sad. I've found out that the manufacturer provides a driver which provides a virtual com port, but still it's windows-only: https://www.ross-tech.com/vag-com/usb/virtual-com-port.php

Maybe the whole topic is a dead-end on Linux. There's a tool called scantool, which can be used with different OBD-II compliant USB devices if they provide a com port ( https://github.com/kees/scantool ). However, I'm not sure if this software requies specific devices (on the scantool.net page I can see that they offer different adapters) and if it is compatible with particular car manufacturers and their cars (like the VAG stuff with its compatibility to Seat, VW, Audi etc. which is the reason why I want to use the adapter and the software...).

If we would like to have support for those adapters, we would need someone who can disassamble the windows driver and create a Linux compatible driver based on his findings...
